topsf/plotting/plot_functions_1d.py

```diff
@@ -101,7 +104,226 @@ def shuffle(entries, mask):
             handles = shuffle(handles, in_stack)
             labels = shuffle(labels, in_stack)
 
-        # make legend using ordered handles/labels
-        ax.legend(handles, labels, **legend_kwargs)
+        pretty_legend = kwargs.pop("pretty_legend")
+        if pretty_legend:
+            import matplotlib.patches as patches
+            import matplotlib.transforms as transforms
+
+            """
+            Define the custom legend entries and their positions:
+            X and Y positions are given in axes fraction
+            ┌──────────────────────────────────────────────────────────────┐
+            │                                                              │
+            │── tt/st merge categories: ───────────────────────────────────│  Y[0]
+            │   ┌─────┐   ┌─────┐                 ┌─────┐                  │
+            │───│     │─/─│     │─── 'annot' ─────│     │─── 'annot' ──────│  Y[1]
+            │   └─────┘   └─────┘                 └─────┘                  │
+            │   :     :   :     :                    :                     │   :
+            │   ┌─────┐   ┌─────┐                    ˌ                     │
+            │───│     │─/─│     │─── 'annot' ─────   o   ─── 'annot' ──────│  Y[4]
+            │   └─────┘   └─────┘                    ˈ                     │
+            └──────────────────────────────────────────────────────────────┘
+                X[1]      X[2]                    X[3]
+                                <--->
+                                 ANNOT_SEP
+            """
+
+            TOP_LABELS = ["fully merged (3q)", "semi-merged (2q)", "not merged (0q or 1q)", "background"]
+            OTHER_LABELS = ["V+jets, VV", "Multijet", "MC stat. unc.", "Data"]
+            X_POSITIONS = [0.05, 0.11, 0.55]  # of (colored) rectangles
+            Y_POSITIONS = [0.9, 0.85, 0.8, 0.75, 0.7]  # of rows
+            RECT_WIDTH = 0.04
+            RECT_HEIGHT = 0.03
+            ANNOT_SEP = 0.05  # annotation separation
+            ANNOTATION_FONT_SIZE = 13
+
+            def add_rectangle(ax, x_pos, y_pos, color, hatch=None):
+                """ Adds a rectangle to the axis with the specified position and color """
+                rect = patches.Rectangle(
+                    (x_pos, y_pos),
+                    RECT_WIDTH,
+                    RECT_HEIGHT,
+                    transform=transforms.blended_transform_factory(ax.transAxes, ax.transAxes),
```

Comment:
IIRC the `blended_transform` is only needed if the `x` and `y` coordinates are of a different type, but if they are both Axes coordinates it should be enough to pass `ax.transAxes` here. 
```suggestion
                    transform=ax.transAxes,
```